src/main/java/apoc/number/exact/Exact.java

```diff
@@ -0,0 +1,99 @@
+	public String sub(@Name("stringA")String stringA, @Name("stringB")String stringB){
+		if(stringA == null || stringA.isEmpty() || stringB == null || stringB.isEmpty()) return null;
+		return new BigDecimal(stringA).subtract(new BigDecimal(stringB)).toPlainString();
+	}
+
+	@UserFunction
+	@Description("apoc.number.exact.mul(stringA,stringB,[prec],[roundingModel]) - return the multiplication's result of two large numbers ")
+	public String mul(@Name("stringA")String stringA, @Name("stringB")String stringB, @Name("precision")Long precision, @Name("roundingMode")String roundingMode){
+		if(stringA == null || stringA.isEmpty() || stringB == null || stringB.isEmpty()) return null;
+		return new BigDecimal(stringA).multiply(new BigDecimal(stringB), getMatchContext(precision, roundingMode)).toPlainString();
+	}
+
+	@UserFunction
+	@Description("apoc.number.exact.div(stringA,stringB,[prec],[roundingModel]) - return the division's result of two large numbers")
+	public String div(@Name("stringA")String stringA, @Name("stringB")String stringB, @Name("precision")Long precision, @Name("roundingMode")String roundingMode){
+		if(stringA == null || stringA.isEmpty() || stringB == null || stringB.isEmpty()) return null;
+		return new BigDecimal(stringA).divide(new BigDecimal(stringB), getMatchContext(precision, roundingMode)).toPlainString();
+	}
+
+	@UserFunction
+	@Description("apoc.number.exact.toInteger(string,[prec],[roundingMode]) - return the Integer value of a large number")
+	public Long toInteger(@Name("stringA")String string, @Name("precision")Long precision, @Name("roundingMode")String roundingMode){
+		if(string == null || string.isEmpty()) return null;
+		return new BigDecimal(string, getMatchContextLong(precision, roundingMode)).longValueExact();
+	}
+
+	@UserFunction
+	@Description("apoc.number.exact.toFloat(string,[prec],[roundingMode]) - return the Float value of a large number")
+	public Double toFloat(@Name("stringA")String string, @Name("precision")Long precision, @Name("roundingMode")String roundingMode){
+		if(string == null || string.isEmpty()) return null;
+		return new BigDecimal(string, getMatchContext(precision, roundingMode)).doubleValue();
+	}
+
+	@UserFunction
+	@Description("apoc.number.exact.toExact(number) - return the exact value")
+	public Long toExact(@Name("number")Long number){
+		if(number == null) return null;
+		return new BigDecimal(number).longValueExact();
+	}
+
+	private MathContext getMatchContext(Long precision, String roundingMode){
+
+		if (precision == null) {
+			precision = Long.valueOf(0);
+		}
+
+		RoundingMode rm = RoundingMode.HALF_UP;
+		if (!StringUtils.isEmpty(roundingMode)) {
+			rm = RoundingMode.valueOf(roundingMode);
+		}
+
+		return new MathContext(precision.intValue(), rm);
+	}
+
+	private MathContext getMatchContextLong(Long precision, String roundingMode){
+		if (precision == null) {
+			precision = Long.valueOf(0);
+		}
+		else {
+			Double pow = pow(10, precision);
```

Comment:
should this be documented? Or do we need a precision at all for turning decimals into Longs?